src/util/tr.js

```diff
@@ -496,75 +499,96 @@ const _tr = (msg, {
       const body = "<div class='wzrkPPdscr' style='color:" + textColor + "'>" + descriptionText + '<div></td></tr></table></div>'
       html = css + title + body
     }
-
-    iframe.setAttribute('style', 'z-index: 2147483647; display:block; width: 100% !important; border:0px !important; border-color:none !important;')
+    // made height : 100% which will help in adjustIframe height
+    iframe.setAttribute('style', 'z-index: 2147483647; display:block; width: 100% !important; height:100%; border:0px !important; border-color:none !important;')
     msgDiv.appendChild(iframe)
-    const ifrm = (iframe.contentWindow) ? iframe.contentWindow : (iframe.contentDocument.document) ? iframe.contentDocument.document : iframe.contentDocument
-    const doc = ifrm.document
 
     // Dispatch event for popup box/banner close
     const closeCampaign = new Event('CT_campaign_rendered')
     document.dispatchEvent(closeCampaign)
 
-    doc.open()
-    doc.write(html)
-
     if (displayObj['custom-editor']) {
-      appendScriptForCustomEvent(targetingMsgJson, doc)
-    }
-    doc.close()
-
-    const adjustIFrameHeight = () => {
-      // adjust iframe and body height of html inside correctly
-      contentHeight = document.getElementById('wiz-iframe').contentDocument.getElementById('contentDiv').scrollHeight
-      if (displayObj['custom-editor'] !== true && !isBanner) {
-        contentHeight += 25
-      }
-      document.getElementById('wiz-iframe').contentDocument.body.style.margin = '0px'
-      document.getElementById('wiz-iframe').style.height = contentHeight + 'px'
+      html = appendScriptForCustomEvent(targetingMsgJson, html)
     }
+    iframe.srcdoc = html
 
     const ua = navigator.userAgent.toLowerCase()
     if (ua.indexOf('safari') !== -1) {
       if (ua.indexOf('chrome') > -1) {
         iframe.onload = () => {
-          adjustIFrameHeight()
-          const contentDiv = document.getElementById('wiz-iframe').contentDocument.getElementById('contentDiv')
-          setupClickUrl(onClick, targetingMsgJson, contentDiv, divId, legacy)
+          if (displayObj['custom-editor']) {
+            iframe.contentWindow.postMessage({
+              action: 'adjustIFrameHeight'
+            }, '*')
+            window.addEventListener('message', event => {
+              if (event.data.action === 'update height') {
+                const heightAdjust = document.getElementById(divId)
+                heightAdjust.style.margin = '0px'
+                heightAdjust.style.height = event.data.value + 'px'
+              }
+            })
+          }
+          setupClickUrl(onClick, targetingMsgJson, '', divId, legacy)
```

Comment:
Have you verified if the clicks get tracked correctly for form based templates? I think it might break coz we are passing an empty string as a part of contentDiv

Reply:
Fixed multiple click tracking issue

Reply:
Added support for form based templates as well